lib/fslib/json/json_mold_namespace_override.cc

```diff
@@ -0,0 +1,398 @@
+enum disir_status
+MoldOverride::retrive_highest_default_version (struct disir_context *context,
+                                               struct semantic_version *highest_semver)
+{
+   enum disir_status status;
+   struct disir_collection *collection;
+   struct disir_context *context_default = NULL;
+   struct semantic_version semver;
+
+   status = dc_get_default_contexts (context, &collection);
+   if (status != DISIR_STATUS_OK)
+   {
+        return status;
+   }
+
+   highest_semver->sv_patch = 0;
+   highest_semver->sv_minor = 0;
+   highest_semver->sv_major = 0;
+
+   // Go through all defaults and determine which
+   // one has the highest version
+   while (dc_collection_next (collection, &context_default)
+          != DISIR_STATUS_EXHAUSTED)
+    {
+        status = dc_get_introduced (context_default, &semver);
+        if (status != DISIR_STATUS_OK)
+            goto out;
+
+        auto eq = dc_semantic_version_compare (&semver, highest_semver);
+        if (eq >= 0)
+        {
+            *highest_semver = semver;
+        }
+        dc_putcontext (&context_default);
+    }
+    // FALL-THROUGH
+out:
+   if (context_default)
+       dc_putcontext (&context_default);
+
+   dc_collection_finished (&collection);
+
+   return status;
+}
+
+enum disir_status
+MoldOverride::set_value_existing_default (struct disir_context *context_keyval,
+                                          struct semantic_version *semver_target,
+                                          Json::Value& value)
+{
+   enum disir_status status;
+   struct disir_collection *collection;
+   struct disir_context *context_default = NULL;
+   struct semantic_version semver;
+
+   status = dc_get_default_contexts (context_keyval, &collection);
+   if (status != DISIR_STATUS_OK)
+   {
+        return status;
+   }
+
+   // Go through all defaults and find a default with matching semver
+   while (dc_collection_next (collection, &context_default)
+          != DISIR_STATUS_EXHAUSTED)
+    {
+        status = dc_get_introduced (context_default, &semver);
+        if (status != DISIR_STATUS_OK)
+            goto out;
+
+        if (dc_semantic_version_compare (&semver, semver_target) == 0)
+        {
+            status = set_value (value, context_default);
+            if (status != DISIR_STATUS_OK)
+                goto out;
+        }
+        dc_putcontext (&context_default);
+    }
+    // FALL-THROUGH
+out:
+   if (context_default)
+       dc_putcontext (&context_default);
+
+   dc_collection_finished (&collection);
+
+   return status;
+}
+
+//! PUBLIC
+enum disir_status
+MoldOverride::apply_overrides (struct disir_context *context_mold)
+{
+    enum disir_status status;
+    struct semantic_version highest_default;
+    struct semantic_version *new_version;
+    struct disir_context *context_keyval = NULL;
+
+    for (const auto& kv : m_override_entries)
+    {
+        status = dc_query_resolve_context (context_mold, kv.first.c_str (), &context_keyval);
```

Comment:
function calls with no parameters shall not have space between function call and opening parenthesis.